packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -260,8 +260,13 @@ export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<v
 }
 
 async function getParameterJson(ctx: SolutionContext) {
+  if (!ctx.targetEnvName) {
+    throw new Error("Failed to get target environment name from solution context.");
+  }
+
+  const parameterFileName = parameterFileNameTemplate.replace("@envName", ctx.targetEnvName);
```

Comment:
Where does the default value `default` configured into `ctx.targetEnvName`?

Reply:
It's in the middleware `envInfoLoader.ts`: https://github.com/OfficeDev/TeamsFx/blob/dev/packages/fx-core/src/core/middleware/envInfoLoader.ts#L55